src/main/java/uk/gov/companieshouse/chipsrestinterfacesconsumer/producer/MessageProducerImpl.java

```diff
@@ -18,35 +19,43 @@
 
     private final ApplicationLogger logger;
 
-    private final AvroSerializer avroSerializer;
+    private final SerializerFactory serializerFactory;
 
     private final CHKafkaProducer producer;
 
     @Autowired
     public MessageProducerImpl(ApplicationLogger logger,
-                               AvroSerializer avroSerializer,
+                               SerializerFactory serializerFactory,
                                CHKafkaProducer producer) {
         this.logger = logger;
-        this.avroSerializer = avroSerializer;
+        this.serializerFactory = serializerFactory;
         this.producer = producer;
     }
 
     @Override
     public void writeToTopic(ChipsRestInterfacesSend chipsMessage, String topicName) throws ServiceException {
         try {
             logger.info(String.format("Writing message id %s to topic: %s", chipsMessage.getMessageId(), topicName));
-            byte[] serializedData = avroSerializer.serialize(chipsMessage);
+            byte[] serializedData = serialize(chipsMessage);
             Message kafkaMessage = new Message();
             kafkaMessage.setValue(serializedData);
             kafkaMessage.setTopic(topicName);
             kafkaMessage.setTimestamp(Long.valueOf(chipsMessage.getCreatedAt()));
             Future<RecordMetadata> future = producer.sendAndReturnFuture(kafkaMessage);
             future.get();
-        } catch (IOException | ExecutionException e) {
+        } catch (ExecutionException e) {
             throw new ServiceException(e.getMessage(), e);
         } catch (InterruptedException ie) {
             Thread.currentThread().interrupt();
             throw new ServiceException("Thread Interrupted when future was sent and returned", ie);
+        } catch (SerializationException e) {
```

Comment:
SonarLint has flagged that SerializationException and ExecutionException branches are identical and can be combined

Reply:
I know.